driver/multi.go

```diff
@@ -0,0 +1,143 @@
+package driver
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/brimsec/zq/ast"
+	"github.com/brimsec/zq/filter"
+	"github.com/brimsec/zq/pkg/nano"
+	"github.com/brimsec/zq/proc"
+	"github.com/brimsec/zq/scanner"
+	"github.com/brimsec/zq/zbuf"
+	"github.com/brimsec/zq/zng/resolver"
+	"go.uber.org/zap"
+)
+
+// A MultiSource is a set of one or more ZNG record sources, which could be
+// a zar archive, a local directory, a collection of remote objects, etc.
+//
+// If the sources are ordered, and that order is expected to be maintained
+// on order-dependent query operations, it must report that ordering
+// via the OrderInfo call, and send SourceOpeners in order.
+type MultiSource interface {
+	// OrderInfo reports if the records for individual sources have an
```

Comment:
I think it would be helpful to future readers to distinguish between inter-source vs intra-source ordering, and which are required (both I believe) to be present in order to return a non-blank `field`.